src/Umbraco.Core/Auditing/AuditEventHandler.cs

```diff
@@ -0,0 +1,284 @@
+                    DateTime.Now,
+                    0, null,
+                    "umbraco/user", "login success");
+            }
+        }
+
+        private void OnLogoutSuccess(object sender, EventArgs args)
+        {
+            if (args is IdentityAuditEventArgs identityArgs)
+            {
+                var performingUser = _userServiceInstance.GetUserById(identityArgs.PerformingUser);
+                _auditServiceInstance.Write(performingUser.Id, $"User \"{performingUser.Name}\" <{performingUser.Email}>", identityArgs.IpAddress,
+                    DateTime.Now,
+                    0, null,
+                    "umbraco/user", "logout success");
+            }
+        }
+
+        private void OnPasswordReset(object sender, EventArgs args)
+        {
+            if (args is IdentityAuditEventArgs identityArgs)
+            {
+                var performingUser = _userServiceInstance.GetUserById(identityArgs.PerformingUser);
+                var affectedUser = _userServiceInstance.GetUserById(identityArgs.AffectedUser);
+                _auditServiceInstance.Write(performingUser.Id, $"User \"{performingUser.Name}\" <{performingUser.Email}>", identityArgs.IpAddress,
+                    DateTime.Now,
+                    affectedUser.Id, $"User \"{affectedUser.Name}\" <{affectedUser.Email}>",
+                    "umbraco/user", "password reset");
+            }
+        }
+
+        private void OnPasswordChanged(object sender, EventArgs args)
+        {
+            if (args is IdentityAuditEventArgs identityArgs)
+            {
+                var performingUser = _userServiceInstance.GetUserById(identityArgs.PerformingUser);
+                var affectedUser = _userServiceInstance.GetUserById(identityArgs.AffectedUser);
+                _auditServiceInstance.Write(performingUser.Id, $"User \"{performingUser.Name}\" <{performingUser.Email}>", identityArgs.IpAddress,
+                    DateTime.Now,
+                    affectedUser.Id, $"User \"{affectedUser.Name}\" <{affectedUser.Email}>",
+                    "umbraco/user", "password change");
+            }
+        }
+
+        private void OnLoginFailed(object sender, EventArgs args)
+        {
+            if (args is IdentityAuditEventArgs identityArgs)
+            {
+                var performingUser = _userServiceInstance.GetUserById(identityArgs.PerformingUser);
+                _auditServiceInstance.Write(performingUser.Id, $"User \"{performingUser.Name}\" <{performingUser.Email}>", identityArgs.IpAddress,
+                    DateTime.Now,
+                    0, null,
+                    "umbraco/user", "login failed");
+            }
+        }
+
+        private void OnForgotPasswordChange(object sender, EventArgs args)
+        {
+            if (args is IdentityAuditEventArgs identityArgs)
+            {
+                var performingUser = _userServiceInstance.GetUserById(identityArgs.PerformingUser);
```

Comment:
Null checks missing on all of these for performingUser and affectedUser ... i'll fix